### DISCUSSION TITLE
How to make a static site generation website using nuxt 3 ? can u show me step by step

### DISCUSSION THREAD

**Discussion by gervasiuswisely** [nuxt/nuxt#24603] [Questions]
- How to make a static site generation website using nuxt 3 ? 
- can u show me step by step to make this in nuxt 3?
-  where to fetch the data is it from the nuxt config or fetch in the pages?
-  and how can i set the data from the api  in pages ? so i can fetch data in pages while generate the pages ?
-  how to generate dynamic routes?
-  should i use nitro.config.prerender().routes?

 thank you so muchh for your help @prakashpie

**Comment by prakashpie** [comment]
@gervasiuswisely I will create a few videos on the same topic and share the link here.

**Reply by gervasiuswisely** [reply]
first of all thank you so much for ur help @prakashpie  i think it will be really big help to everyone too and i found a bug or error when i run the result of npm run generate when i open page that have fetch api it shows that it gets _payload.json but when i refresh the pages it run the _nuxt/index.f63002c8.js which is not the right url im using usingFetch for the fetch api by the way because it can give us the data to _payload.json. additional question what the use of the properties key in my useFetch function. thank you so much for your help @prakashpie 👍

**Reply by prakashpie** [reply]
Hi @gervasiuswisely 

The payload feature was not available in Nuxt versions below v3.10.xx. However, it is now available in Nuxt v3.10 as an experimental feature. I have updated the code for both scenarios, with payloads and without payloads.

Please check this repository:

[nuxt3-dynamic-routes](https://github.com/codepie-io/nuxt3-dynamic-routes)

I hope it will be helpful for you.
